File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/ErdOnlineProjectService.java
package org.javamaster.invocationlab.admin.service;

import org.javamaster.invocationlab.admin.enums.ProjectType;
import org.javamaster.invocationlab.admin.enums.RoleGroupEnum;
import org.javamaster.invocationlab.admin.model.erd.DbsBean;
import org.javamaster.invocationlab.admin.model.erd.ErdOnlineModel;
import org.javamaster.invocationlab.admin.model.erd.GroupGetVo;
import org.javamaster.invocationlab.admin.model.erd.ModulesBean;
import org.javamaster.invocationlab.admin.model.erd.PageVo;
import org.javamaster.invocationlab.admin.model.erd.RecordsVo;
import org.javamaster.invocationlab.admin.model.erd.SaveProjectVo;
import org.javamaster.invocationlab.admin.model.erd.SortModuleReqVo;
import org.javamaster.invocationlab.admin.model.erd.StatisticVo;
import org.javamaster.invocationlab.admin.model.erd.TokenVo;
import org.javamaster.invocationlab.admin.model.erd.UsersVo;
import com.alibaba.fastjson.JSONObject;
import org.apache.commons.lang3.tuple.Pair;

import java.util.List;

/**
 * @author yudong
 */
public interface ErdOnlineProjectService {

    StatisticVo statistic() throws Exception;

    PageVo getProjectList(TokenVo tokenVo, ProjectType projectType, String projectName) throws Exception;

    void saveRecordsVoToGroupUsers(String projectId, List<UsersVo> reqUsersVos, RoleGroupEnum roleGroupEnum, TokenVo tokenVo) throws Exception;

    void delRecordsVoFromGroupUsers(String projectId, List<UsersVo> reqUsersVos, TokenVo tokenVo) throws Exception;

    RecordsVo findRecordsVoByProjectId(String projectId, TokenVo tokenVo) throws Exception;

    ErdOnlineModel getProjectDetail(String projectId, TokenVo tokenVo) throws Exception;

    void eraseSensitiveInfo(ErdOnlineModel erdOnlineModel, String userId);

    void resumeSensitiveInfo(ErdOnlineModel erdOnlineModel, String userId);

    DbsBean getDefaultDb(String projectId, TokenVo tokenVo) throws Exception;

    String createProject(JSONObject jsonObjectReq, TokenVo tokenVo) throws Exception;

    String deleteProject(JSONObject jsonObjectReq, TokenVo tokenVo) throws Exception;

    String updateProjectBasicInfo(RecordsVo recordsVoReq, TokenVo tokenVo) throws Exception;

    Boolean saveProject(SaveProjectVo saveProjectVo, TokenVo tokenVo) throws Exception;

    ModulesBean refreshProjectModule(JSONObject jsonObjectReq, TokenVo tokenVo) throws Exception;

    GroupGetVo getProjectBasicInfo(String projectId, TokenVo tokenVo) throws Exception;

    Pair<String, byte[]> exportErd(JSONObject jsonObjectReq, TokenVo tokenVo) throws Exception;

    String sortModule(SortModuleReqVo reqVo, TokenVo tokenVo) throws Exception;
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/ErdOnlineProjectService.kt
package org.javamaster.invocationlab.admin.service

import org.javamaster.invocationlab.admin.enums.ProjectType
import org.javamaster.invocationlab.admin.enums.RoleGroupEnum
import org.javamaster.invocationlab.admin.model.erd.*
import com.alibaba.fastjson.JSONObject
import org.apache.commons.lang3.tuple.Pair

/**
 * @author yudong
 */
interface ErdOnlineProjectService {

    fun statistic(): StatisticVo

    fun getProjectList(tokenVo: TokenVo, projectType: ProjectType, projectName: String?): PageVo

    fun saveRecordsVoToGroupUsers(
        projectId: String,
        reqUsersVos: List<UsersVo>,
        roleGroupEnum: RoleGroupEnum,
        tokenVo: TokenVo
    )

    fun delRecordsVoFromGroupUsers(projectId: String, reqUsersVos: List<UsersVo>, tokenVo: TokenVo)

    fun findRecordsVoByProjectId(projectId: String, tokenVo: TokenVo): RecordsVo

    fun getProjectDetail(projectId: String, tokenVo: TokenVo): ErdOnlineModel

    fun eraseSensitiveInfo(erdOnlineModel: ErdOnlineModel, userId: String)

    fun resumeSensitiveInfo(erdOnlineModel: ErdOnlineModel, userId: String)

    fun getDefaultDb(projectId: String, tokenVo: TokenVo): DbsBean

    fun createProject(jsonObjectReq: JSONObject, tokenVo: TokenVo): String

    fun deleteProject(jsonObjectReq: JSONObject, tokenVo: TokenVo): String

    fun updateProjectBasicInfo(recordsVoReq: RecordsVo, tokenVo: TokenVo): String

    fun saveProject(saveProjectVo: SaveProjectVo, tokenVo: TokenVo): Boolean

    fun refreshProjectModule(jsonObjectReq: JSONObject, tokenVo: TokenVo): ModulesBean

    fun getProjectBasicInfo(projectId: String, tokenVo: TokenVo): GroupGetVo

    fun exportErd(jsonObjectReq: JSONObject, tokenVo: TokenVo): Pair<String, ByteArray>

    fun sortModule(reqVo: SortModuleReqVo, tokenVo: TokenVo): String
}
